### PR TITLE
test: do not test if snprintf() fails in rpma_log_default_function()

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/rpma/297)

### CODE REVIEW

**Review by janekmi — CHANGES_REQUESTED** [review]
Great job! :+1: 

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFgUh518sDVzAZize84:b-5w53r2)* status: all files reviewed, 1 unresolved discussion (waiting on @ldorau)

___
*[examples/04-write-to-persistent/client.c, line 42 at r1](https://reviewable.io/reviews/pmem/rpma/297#-MFgRo35EXXemvm_929U:-MFgRo35EXXemvm_929V:b-dgaubw) ([raw file](https://github.com/pmem/rpma/blob/0106a99aef847099a94ba3f7cb1d9cc54826a93c/examples/04-write-to-persistent/client.c#L42)):*
> ```C
> {
> 	hello->lang = lang;
> 	strncpy(hello->str, hello_str[hello->lang], KILOBYTE);
> ```

**Very picky**: The server in this example prepares a kilobyte for this data on the opposite site. Assuming the buffer on the client site will be filled completely (1KiB of non-zero characters + 1 null termination character), on the server site it will be a not null-terminated string. I know the strings in this example are very short comparing to 1KiB. Sorry. I couldn't resist.

**TLDR**: I think the right solution is `KILOBYTE - 1` in this line.

**Review by ldorau — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: 2 of 3 files reviewed, 1 unresolved discussion (waiting on @janekmi)

___
*[examples/04-write-to-persistent/client.c, line 42 at r1](https://reviewable.io/reviews/pmem/rpma/297#-MFgRo35EXXemvm_929U:-MFgVwOw9cr55KTElOrv:bizp97x) ([raw file](https://github.com/pmem/rpma/blob/0106a99aef847099a94ba3f7cb1d9cc54826a93c/examples/04-write-to-persistent/client.c#L42)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

**Very picky**: The server in this example prepares a kilobyte for this data on the opposite site. Assuming the buffer on the client site will be filled completely (1KiB of non-zero characters + 1 null termination character), on the server site it will be a not null-terminated string. I know the strings in this example are very short comparing to 1KiB. Sorry. I couldn't resist.

**TLDR**: I think the right solution is `KILOBYTE - 1` in this line.
</blockquote></details>

Right :-). Done.

**Review by grom72 — COMMENTED** [review]
:goat: 

> Reviewed 2 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFj8hpCCsUJRLk9PA9N:bdycpsm)* status: all files reviewed, 2 unresolved discussions (waiting on @janekmi and @ldorau)

___
*[src/log_default.c, line 45 at r2](https://reviewable.io/reviews/pmem/rpma/297#-MFj876v1TGmZ0ddeSh_:-MFj876v1TGmZ0ddeSha:bvcbn7b) ([raw file](https://github.com/pmem/rpma/blob/26a334360e53fcc9159138fff0e8ca4fac003b3f/src/log_default.c#L45)):*
> ```
> get_timestamp_prefix
> ```
add rpma_ prefix to avoid any conflict in future

**Review by grom72 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 4 unresolved discussions (waiting on @janekmi and @ldorau)

___
*[src/log_default.c, line 41 at r2](https://reviewable.io/reviews/pmem/rpma/297#-MFj8s9eAM7pdEHJ-Ws5:-MFj8s9eAM7pdEHJ-Ws6:b2ggvxm) ([raw file](https://github.com/pmem/rpma/blob/8167ca0dca6595c5f703f380f89a41234f03afb7/src/log_default.c#L41)):*
> ```C
>  * - buf != NULL && buf_size >= 16
>  */
> static void
> ```

can we do this only 
___
*[src/log_default.c, line 44 at r2](https://reviewable.io/reviews/pmem/rpma/297#-MFj8vPm5eGcWf5oVq_i:-MFj8vPm5eGcWf5oVq_j:b4dwfqf) ([raw file](https://github.com/pmem/rpma/blob/26a334360e53fcc9159138fff0e8ca4fac003b3f/src/log_default.c#L44)):*
> ```
> void
> ```
can we wraf this with #if _non debug_ ...

**Review by ldorau — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: 2 of 4 files reviewed, 4 unresolved discussions (waiting on @grom72 and @janekmi)

___
*[src/log_default.c, line 41 at r2](https://reviewable.io/reviews/pmem/rpma/297#-MFj8s9eAM7pdEHJ-Ws5:-MFjCibp77sglusYh0ac:bpfrlxh) ([raw file](https://github.com/pmem/rpma/blob/8167ca0dca6595c5f703f380f89a41234f03afb7/src/log_default.c#L41)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

can we do this only 
</blockquote></details>

? I do not understand what is your point?
___
*[src/log_default.c, line 44 at r2](https://reviewable.io/reviews/pmem/rpma/297#-MFj8vPm5eGcWf5oVq_i:-MFjDOc_3V-Kd2vOLtd9:b9w6p91) ([raw file](https://github.com/pmem/rpma/blob/26a334360e53fcc9159138fff0e8ca4fac003b3f/src/log_default.c#L44)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

> ```
> void
> ```
can we wraf this with #if _non debug_ ...
</blockquote></details>

Why? It would add an unneeded complication to the code.
___
*[src/log_default.c, line 45 at r2](https://reviewable.io/reviews/pmem/rpma/297#-MFj876v1TGmZ0ddeSh_:-MFjDaxi2Ok9Wt-HBuCJ:b-896fix) ([raw file](https://github.com/pmem/rpma/blob/26a334360e53fcc9159138fff0e8ca4fac003b3f/src/log_default.c#L45)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

> ```
> get_timestamp_prefix
> ```
add rpma_ prefix to avoid any conflict in future
</blockquote></details>

Done.

**Review by grom72 — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 3 files at r1, 1 of 2 files at r3, 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFjEHbh1r2gWWONtmeE:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @janekmi)

**Review by grom72 — COMMENTED** [review]
after r7 - I do not see any additional value to keep a separate set of test only for DEBUG
especially that we test something that is "almost" impossible. are we talk only about  codecove?
perhaps we should explicitly ignore result in snprintf inside get_timestamp_prefix
or
return uS separately and add this to the message in rpma_log_default_function function

> Reviewed 1 of 2 files at r6, 2 of 2 files at r7.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFkASlv1-QYuDiKW-JP:b439d7q)* status: all files reviewed, 4 unresolved discussions (waiting on @janekmi and @ldorau)

___
*[tests/unit/log_default/function.c, line 75 at r6](https://reviewable.io/reviews/pmem/rpma/297#-MFk95dC4xT7mwuWnbdg:-MFk95dC4xT7mwuWnbdh:b-c3c0q5) ([raw file](https://github.com/pmem/rpma/blob/516d604d48567d8f77332fe5b0befedd98ecc806/tests/unit/log_default/function.c#L75)):*
> ```
> int snprintf_error;
> ```
do we need this? - this was added to r6
___
*[tests/unit/log_default/function.c, line 213 at r6](https://reviewable.io/reviews/pmem/rpma/297#-MFk9_gg-HnN0iEvBZpV:-MFk9_gg-HnN0iEvBZpW:b-5j9tcj) ([raw file](https://github.com/pmem/rpma/blob/516d604d48567d8f77332fe5b0befedd98ecc806/tests/unit/log_default/function.c#L213)):*
> ```
> else if ((x)->snprintf_error)
> ```
do we need this at all if we decided not to emulate snprinft error at all?
snprintf_error is always == 0 based on r6 version
___
*[tests/unit/log_default/function.c, line 369 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFkAFzBC-94_5WFJbS2:-MFkAFzC7H5CV4oo-5-5:bygjltu) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/tests/unit/log_default/function.c#L369)):*
> ```
> DEBUG
> ```
what will be added value of this test only in DEBUG ? the library shall be released as RELEASE always

**Review by ldorau — COMMENTED** [review]
Code coverage only

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFkD1MP5K0fgQBDboEx:baccq8m)* status: all files reviewed, 4 unresolved discussions (waiting on @grom72 and @janekmi)

___
*[tests/unit/log_default/function.c, line 75 at r6](https://reviewable.io/reviews/pmem/rpma/297#-MFk95dC4xT7mwuWnbdg:-MFkCaHc5PXGmm7VZmyB:bsk1g43) ([raw file](https://github.com/pmem/rpma/blob/516d604d48567d8f77332fe5b0befedd98ecc806/tests/unit/log_default/function.c#L75)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

> ```
> int snprintf_error;
> ```
do we need this? - this was added to r6
</blockquote></details>

It has been already there before. It is still needed if we want to run this test in the Debug build.
___
*[tests/unit/log_default/function.c, line 213 at r6](https://reviewable.io/reviews/pmem/rpma/297#-MFk9_gg-HnN0iEvBZpV:-MFkCqwb6gBMMcwPTR1C:b3kqx) ([raw file](https://github.com/pmem/rpma/blob/516d604d48567d8f77332fe5b0befedd98ecc806/tests/unit/log_default/function.c#L213)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

> ```
> else if ((x)->snprintf_error)
> ```
do we need this at all if we decided not to emulate snprinft error at all?
snprintf_error is always == 0 based on r6 version
</blockquote></details>

.
___
*[tests/unit/log_default/function.c, line 369 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFkAFzBC-94_5WFJbS2:-MFkCuIhEm5cjHo5UWrl:baccq8m) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/tests/unit/log_default/function.c#L369)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

> ```
> DEBUG
> ```
what will be added value of this test only in DEBUG ? the library shall be released as RELEASE always
</blockquote></details>

Code coverage only

**Review by grom72 — COMMENTED** [review]
so I would propose to use (void) snprinft(...) inside get_timestamp_prefix

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFkcnPu8UvtksWGGT4H:b7593tz)* status: all files reviewed, 4 unresolved discussions (waiting on @janekmi and @ldorau)

___
*[tests/unit/log_default/function.c, line 75 at r6](https://reviewable.io/reviews/pmem/rpma/297#-MFk95dC4xT7mwuWnbdg:-MFkdD_-6ReSbRd-OGhK:btye78g) ([raw file](https://github.com/pmem/rpma/blob/516d604d48567d8f77332fe5b0befedd98ecc806/tests/unit/log_default/function.c#L75)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

It has been already there before. It is still needed if we want to run this test in the Debug build.
</blockquote></details>

I do not want to test OS or libc. It does not help especially that this is a part of log which should not stop librpma in runtime.
___
*[tests/unit/log_default/function.c, line 213 at r6](https://reviewable.io/reviews/pmem/rpma/297#-MFk9_gg-HnN0iEvBZpV:-MFkdceQ-JO0L1JXhVbB:b3kqx) ([raw file](https://github.com/pmem/rpma/blob/516d604d48567d8f77332fe5b0befedd98ecc806/tests/unit/log_default/function.c#L213)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

.
</blockquote></details>

.
___
*[tests/unit/log_default/function.c, line 369 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFkAFzBC-94_5WFJbS2:-MFkd3bG0u-OLXu9BMyB:b-20x3g8) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/tests/unit/log_default/function.c#L369)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Code coverage only
</blockquote></details>

so if we (void) snprinf then the problem disappears

**Review by ldorau — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 4 unresolved discussions (waiting on @janekmi and @ldorau)

___
*[tests/unit/log_default/function.c, line 369 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFkAFzBC-94_5WFJbS2:-MFnlDZh6RWpL_s_Ej8h:bsdxhuk) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/tests/unit/log_default/function.c#L369)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

so if we (void) snprinf then the problem disappears
</blockquote></details>

Are we **100% SURE** that this snprinf() call will **NEVER** fail?
What if something changes and the compiler stops to optimize out the failing path?
@janekmi what do you think ?

**Review by ldorau — COMMENTED** [review]
Are we **100% SURE** that this snprinf() call will **NEVER** fail?
What if something changes and the compiler stops to optimize out the failing path?
@janekmi what do you think ?

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFnlw3SEmxEXXtY4VuD:bsdxhuk)* status: all files reviewed, 4 unresolved discussions (waiting on @janekmi and @ldorau)

**Review by grom72 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 4 unresolved discussions (waiting on @janekmi and @ldorau)

___
*[tests/unit/log_default/function.c, line 369 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFkAFzBC-94_5WFJbS2:-MFnmWEbDlPjFMZr-XjY:b-lcwish) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/tests/unit/log_default/function.c#L369)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Are we **100% SURE** that this snprinf() call will **NEVER** fail?
What if something changes and the compiler stops to optimize out the failing path?
@janekmi what do you think ?
</blockquote></details>

I'm ok to leave this test in case we disable optimization for Release version => add_flag("-U_FORTIFY_SOURCE" RELEASE)

**Review by ldorau — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 4 unresolved discussions (waiting on @janekmi and @ldorau)

___
*[tests/unit/log_default/function.c, line 369 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFkAFzBC-94_5WFJbS2:-MFnpKo375vza1ZRcfx_:b-h9dz71) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/tests/unit/log_default/function.c#L369)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

I'm ok to leave this test in case we disable optimization for Release version => add_flag("-U_FORTIFY_SOURCE" RELEASE)
</blockquote></details>

`_FORTIFY_SOURCE` has nothing to do with this issue.

**Review by grom72 — COMMENTED** [review]
my answer is in #305 

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFoDXlzDEzBUNgJ0L_R:b-tzei6l)* status: all files reviewed, 1 unresolved discussion (waiting on @janekmi)

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 1 unresolved discussion (waiting on @janekmi)

*[a discussion](https://reviewable.io/reviews/pmem/rpma/297#-MFpqqDJAW7J2o5VEeMo:-MFpqqDJAW7J2o5VEeMp:br4ems0) (no related file):*
This PR is a great example that the rule: one PR == one change is very important. Two commits from this PR are ready for merge since 2 days now but the discussion around the third one blocks them from being merged.

---

**Review by janekmi — CHANGES_REQUESTED** [review]
In my opinion, #305 proves the snprintf, the way it is used here, simply can not fail. I do not think writing snprintf subset on our own is the solution here.

@grom72 if you agree to fix in the proposed way please close #305.

> Reviewed 1 of 3 files at r2, 2 of 2 files at r7.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFptz0X0omYekAso-kJ:bcneb26)* status: all files reviewed, 3 unresolved discussions (waiting on @ldorau)

___
*[src/log_default.c, line 46 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFptMXA8bFPpslkB5_q:-MFptMXA8bFPpslkB5_r:b-o2frrn) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/src/log_default.c#L46)):*
> ```C
>  * When this function stops being static, the unit test
>  * for the failing path of the snprintf() function should be enabled
>  * in case of the Release build.
> ```

NOTE
This function is static now, so we know all possible calls of the snprintf and conclude it can not fail.
___
*[src/log_default.c, line 73 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFpsK5E7urx1K5tFknV:-MFpsK5F7hfwGXvbL6hF:bvapkk2) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/src/log_default.c#L73)):*
> ```C
> 	}
> 
> 	if (snprintf(buf, buf_size, "[%s.%06ld] ", date, usec) < 0) {
> ```

```c
/* It can not fail. Please see the note above. */
(void) snprintf(/* ... */);
```
___
*[tests/unit/log_default/function.c, line 334 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFptfsq0_y34yndlikk:-MFptfsr3cpx0C8yf00c:b-dstzi1) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/tests/unit/log_default/function.c#L334)):*
> ```C
> 	0, 0, 0, 1, RPMA_LOG_LEVEL_DEBUG, MOCK_FILE_NAME
> };
> #endif
> ```

Please remove this test case.

**Review by grom72 — COMMENTED** [review]
we do not need snprintf to convert just one number into six-digit string ltoa return void but snprintf returns int and if you look at it in details it is very heavy one

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFqY0yX7kbUVfV7CT45:b-lqm2qn)* status: all files reviewed, 3 unresolved discussions (waiting on @ldorau)

**Review by grom72 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 3 unresolved discussions (waiting on @janekmi and @ldorau)

___
*[src/log_default.c, line 73 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFpsK5E7urx1K5tFknV:-MFqYuQ099yE80Jyul-m:bcfq8ej) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/src/log_default.c#L73)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

```c
/* It can not fail. Please see the note above. */
(void) snprintf(/* ... */);
```
</blockquote></details>

I agree to use void, the only problem here is performance.
There #305  is implemented in an optimize way - no copy of data from one to another one. No using HUGE snprintf just to convert number to string as it is done by itoa() C function.

**Review by grom72 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 3 unresolved discussions (waiting on @janekmi and @ldorau)

___
*[src/log_default.c, line 46 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFptMXA8bFPpslkB5_q:-MFqZVbKC8aFWgC5ezNv:bionvw4) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/src/log_default.c#L46)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

NOTE
This function is static now, so we know all possible calls of the snprintf and conclude it can not fail.
</blockquote></details>

But as I mentioned before we do not know if this will be also true in the future. We do not have any test that protects us from a situation when this function might fail in the future.

**Review by grom72 — COMMENTED** [review]
and what about #292? do we have some tests running only in Debug?

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MFq_12T5dicXmaE3B19:bu701vg)* status: all files reviewed, 3 unresolved discussions (waiting on @janekmi and @ldorau)

**Review by ldorau — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 3 unresolved discussions (waiting on @janekmi and @ldorau)

*[a discussion](https://reviewable.io/reviews/pmem/rpma/297#-MFpqqDJAW7J2o5VEeMo:-MG2TY7g5izL1w-As481:b-5f3kq5) (no related file):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

This PR is a great example that the rule: one PR == one change is very important. Two commits from this PR are ready for merge since 2 days now but the discussion around the third one blocks them from being merged.
</blockquote></details>

I moved those 2 commits to https://github.com/pmem/rpma/pull/315

---

**Review by ldorau — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 3 unresolved discussions (waiting on @janekmi)

___
*[src/log_default.c, line 46 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFptMXA8bFPpslkB5_q:-MG2_oRO5vY3U47384kM:b-896fix) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/src/log_default.c#L46)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

But as I mentioned before we do not know if this will be also true in the future. We do not have any test that protects us from a situation when this function might fail in the future.
</blockquote></details>

Done.
___
*[src/log_default.c, line 73 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFpsK5E7urx1K5tFknV:-MG2boe_1HCaxd1sbV38:b-896fix) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/src/log_default.c#L73)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

I agree to use void, the only problem here is performance.
There #305  is implemented in an optimize way - no copy of data from one to another one. No using HUGE snprintf just to convert number to string as it is done by itoa() C function.   
</blockquote></details>

Done.
___
*[tests/unit/log_default/function.c, line 334 at r7](https://reviewable.io/reviews/pmem/rpma/297#-MFptfsq0_y34yndlikk:-MG2dqeO9NIIlOL-pdeA:b-896fix) ([raw file](https://github.com/pmem/rpma/blob/353f60bb2b4c2fda76ea5f11b7ae924d0158c612/tests/unit/log_default/function.c#L334)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

Please remove this test case.
</blockquote></details>

Done.

**Review by janekmi — CHANGES_REQUESTED** [review]
> Reviewed 2 of 4 files at r8.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: all files reviewed, 1 unresolved discussion (waiting on @ldorau)

___
*[src/log_default.c, line 69 at r8](https://reviewable.io/reviews/pmem/rpma/297#-MG2g_RzA2JxpcSTEJUr:-MG2g_RzA2JxpcSTEJUs:b4eig43) ([raw file](https://github.com/pmem/rpma/blob/b726be4fd188b0f46f9dda144766c31439069da8/src/log_default.c#L69)):*
> ```C
> 	/* it cannot fail - please see the note above */
> 	(void) snprintf(buf, buf_size, "[%s.%06ld] ", date, usec);
> 	if (strlen(buf) == buf_size)
> ```

strNlen

**Review by ldorau — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @janekmi)

___
*[src/log_default.c, line 69 at r8](https://reviewable.io/reviews/pmem/rpma/297#-MG2g_RzA2JxpcSTEJUr:-MG2hY0G7DN4JpVjd8Dp:b-896fix) ([raw file](https://github.com/pmem/rpma/blob/b726be4fd188b0f46f9dda144766c31439069da8/src/log_default.c#L69)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

strNlen
</blockquote></details>

Done.

**Review by janekmi — APPROVED** [review]
:lgtm:

> Reviewed 1 of 3 files at r9.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MG2i6ZWBCZrR6rHOM59:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by grom72 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MG2lHjR6yzF5cd8Eqj9:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

___
*[src/log_default.c, line 72 at r9](https://reviewable.io/reviews/pmem/rpma/297#-MG2l4uz97V6lroOtQy6:-MG2l4uz97V6lroOtQy7:b-51yxy) ([raw file](https://github.com/pmem/rpma/blob/ee6bbb13a5502321e42398d2fc104605a1a6fcf4/src/log_default.c#L72)):*
> ```
> strnlen
> ```
very heavy solution ... 
https://code.woboq.org/userspace/glibc/string/strnlen.c.html
but seems I have no option other than accept this :(

**Review by grom72 — COMMENTED** [review]
what about mock_config adjustment to remove snprintf reference there?

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MG2lWrd1sr1C4Qk2fAm:b-8xnsu3)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by janekmi — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: :shipit: complete! all files reviewed, all discussions resolved

___
*[src/log_default.c, line 72 at r9](https://reviewable.io/reviews/pmem/rpma/297#-MG2l4uz97V6lroOtQy6:-MG2mR_DB8NEA_AbXdZA:b1gltw2) ([raw file](https://github.com/pmem/rpma/blob/ee6bbb13a5502321e42398d2fc104605a1a6fcf4/src/log_default.c#L72)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

> ```
> strnlen
> ```
very heavy solution ... 
https://code.woboq.org/userspace/glibc/string/strnlen.c.html
but seems I have no option other than accept this :(
</blockquote></details>

It looks complicated. But it is complicated because it is heavily optimized for the sake of performance. ;-)

**Review by ldorau — COMMENTED** [review]
Replaced with `snprintf_no_eol`

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MG33lvM4RJbtJxjfYg-:byqjio2)* status: 1 of 3 files reviewed, all discussions resolved (waiting on @janekmi)

**Review by ldorau — COMMENTED** [review]
Merging #297 into master **will not change coverage**. The diff coverage is **100.00%**.

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MG34gTvBisEWBybSdmp:brhvah6)* status: 1 of 3 files reviewed, all discussions resolved (waiting on @janekmi)

**Review by janekmi — CHANGES_REQUESTED** [review]
Great news! :+1: 

> Reviewed 2 of 2 files at r10.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MG2px8w8yybjcl2HwuC:b-mczwp0)* status: all files reviewed, 2 unresolved discussions (waiting on @ldorau)

___
*[tests/unit/common/mocks-stdio.c, line 39 at r10](https://reviewable.io/reviews/pmem/rpma/297#-MG3AtI96vIHUl9_YgaG:-MG3AtI96vIHUl9_YgaH:b-l7lrdi) ([raw file](https://github.com/pmem/rpma/blob/035fb97f80ba8151f2491ad9e1feec9bd2a5ec43/tests/unit/common/mocks-stdio.c#L39)):*
> ```C
> 	if (ret < 0)
> 		return ret;
> 	if (ret > 0) {
> ```

ret == MOCK_SNPRINTF_NO_EOL 
___
*[tests/unit/log_default/function.c, line 194 at r10](https://reviewable.io/reviews/pmem/rpma/297#-MG3AZEB4iU5V0BAxeVH:-MG3AZEB4iU5V0BAxeVI:b-lgv0l) ([raw file](https://github.com/pmem/rpma/blob/035fb97f80ba8151f2491ad9e1feec9bd2a5ec43/tests/unit/log_default/function.c#L194)):*
> ```C
> #define MOCK_TIME_STR "[" MOCK_TIME_OF_DAY_STR ".000000] "
> #define MOCK_TIME_ERROR_STR "[time error] "
> #define MOCK_SNPRINTF_NO_EOL 1 /* any >0 value */
> ```

Please move it to mocks-stdio.h in the following form:

```c
#define MOCK_SNPRINTF_NO_EOL MAX_SIZE
```

**Review by ldorau — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: 1 of 4 files reviewed, 2 unresolved discussions (waiting on @janekmi)

___
*[tests/unit/common/mocks-stdio.c, line 39 at r10](https://reviewable.io/reviews/pmem/rpma/297#-MG3AtI96vIHUl9_YgaG:-MG3BE9b1VgWyq6aojFW:b-896fix) ([raw file](https://github.com/pmem/rpma/blob/035fb97f80ba8151f2491ad9e1feec9bd2a5ec43/tests/unit/common/mocks-stdio.c#L39)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

ret == MOCK_SNPRINTF_NO_EOL 
</blockquote></details>

Done.
___
*[tests/unit/log_default/function.c, line 194 at r10](https://reviewable.io/reviews/pmem/rpma/297#-MG3AZEB4iU5V0BAxeVH:-MG3BHpp4xRWxF7aUdAB:b-896fix) ([raw file](https://github.com/pmem/rpma/blob/035fb97f80ba8151f2491ad9e1feec9bd2a5ec43/tests/unit/log_default/function.c#L194)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

Please move it to mocks-stdio.h in the following form:

```c
#define MOCK_SNPRINTF_NO_EOL MAX_SIZE
```
</blockquote></details>

Done.

**Review by janekmi — APPROVED** [review]
> Reviewed 3 of 3 files at r11.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by janekmi — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MG3Dbd_B1k4757qAwis:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by grom72 — COMMENTED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/297#-:-MG3DSth09PLJL7lErpC:bg7igkv)* status: :shipit: complete! all files reviewed, all discussions resolved
